src/zenml/post_execution/pipeline.py

```diff
@@ -123,9 +123,9 @@ def get_pipeline(
         name=pipeline_name,
         project_name_or_id=active_project_id,
     )
-    if len(pipeline_models) == 1:
+    if pipeline_models.total == 1:
```

Comment:
This change made me think that it might be a good idea to implement the `__len__` method on that class so that we can use the previous syntax?

Reply:
The question here is would you expect `len(Page)` to return the total number of items available, or the number of items in that specific page? Probably the number of items in the page, right? In that case, using `len` for this check could lead to issues if users set the page size limit to 1 for some reason.

Reply:
I added a couple of container methods to `Page` so we can now do `page[0]`, `x in page`, `len(page)`, `[a for a in page]`

Reply:
Yeah I was thinking of just getting the amount of items in that page. But you're right, the `total` property is still needed and also the right one to use in this case.